src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -209,8 +209,16 @@ public TopicPartitionWriter(
         config.getString(StorageSinkConnectorConfig.SCHEMA_COMPATIBILITY_CONFIG));
 
     String logsDir = config.getLogsDirFromTopic(tp.topic());
-    boolean disableWal = config.getBoolean(HdfsSinkConnectorConfig.DISABLE_WAL_CONFIG);
-    wal = disableWal ? new NoopWAL() : storage.wal(logsDir, tp);
+
+    String walType = config.getString(HdfsSinkConnectorConfig.WAL_TYPE);
+
+    if (walType.equalsIgnoreCase(HdfsSinkConnectorConfig.WAL_NOOP)) {
+      wal = new NoopWAL();
+    } else if (walType.equalsIgnoreCase(HdfsSinkConnectorConfig.WAL_QFS)) {
+      wal = storage.wal(logsDir, tp, true);
```

Comment:
I'd probably add some explicit public methods qfsWal() and hdfsWal() instead of passing true/false